Capri.Services/Courses/CourseCreator.cs

```diff
@@ -0,0 +1,52 @@
+using System.Threading.Tasks;
+using AutoMapper;
+using Capri.Database;
+using Capri.Database.Entities;
+using Capri.Services.Faculties;
+using Capri.Web.ViewModels.Course;
+
+namespace Capri.Services.Courses
+{
+    public class CourseCreator : ICourseCreator
+    {
+        private readonly ISqlDbContext _context;
+        private readonly IMapper _mapper;
+        private readonly IFacultyGetter _facultyGetter;
+
+        public CourseCreator(
+            ISqlDbContext context,
+            IMapper mapper,
+            IFacultyGetter facultyGetter
+        )
+        {
+            _context = context;
+            _mapper = mapper;
+            _facultyGetter = facultyGetter;
+        }
+
+        public async Task<IServiceResult<CourseViewModel>> Create(CourseRegistration registration)
+        {
+            var facultyResult = await _facultyGetter.Get(registration.FacultyId);
+            if(!facultyResult.Successful())
+            {
+                return ServiceResult<CourseViewModel>.Error(facultyResult.GetAggregatedErrors());
+            }
+
+            var course = _mapper.Map<Course>(registration);
+
+            try 
+            {
```

Comment:
Why you use that try? What type of Exception you expect here cause I don't see any type below.

Reply:
I expect invalid data. Unless we validated data before calling this method